File: rust/src/leetcode/Q268_missing_number.rs
/*
 * @lc app=leetcode id=268 lang=rust
 *
 * [268] Missing Number
 */

// @lc code=start
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let mut result: i32 = nums.len() as i32;

        for (index, number) in nums.into_iter().enumerate() {
            result ^= index as i32 ^ number;
        }

        result
    }
}
// @lc code=end

File: python/Q268_missing_number.py
#
# @lc app=leetcode id=268 lang=python3
#
# [268] Missing Number
#


# @lc code=start
class Solution:
    def missingNumber(self, nums: list[int]) -> int:
        result: int = len(nums)

        for i, v in enumerate(nums):
            result ^= i ^ v

        return result


# @lc code=end
